config/config_test.go

```diff
@@ -34,6 +35,91 @@ func TestT(t *testing.T) {
 	TestingT(t)
 }
 
+func (s *testConfigSuite) TestNullableBoolUnmashal(c *C) {
+	var nb = nullableBool{false, false}
+	data, err := json.Marshal(nb)
+	c.Assert(err, IsNil)
+	err = json.Unmarshal(data, &nb)
+	c.Assert(err, IsNil)
+	c.Assert(nb, Equals, nbUnset)
+
+	nb = nullableBool{true, false}
+	data, err = json.Marshal(nb)
+	c.Assert(err, IsNil)
+	err = json.Unmarshal(data, &nb)
+	c.Assert(err, IsNil)
+	c.Assert(nb, Equals, nbFalse)
+
+	nb = nullableBool{true, true}
+	data, err = json.Marshal(nb)
+	c.Assert(err, IsNil)
+	err = json.Unmarshal(data, &nb)
+	c.Assert(err, IsNil)
+	c.Assert(nb, Equals, nbTrue)
+}
+
+func (s *testConfigSuite) TestLogConfig(c *C) {
+	var conf Config
+	configFile := "log_config.toml"
+	_, localFile, _, _ := runtime.Caller(0)
+	configFile = path.Join(path.Dir(localFile), configFile)
+
+	f, err := os.Create(configFile)
+	c.Assert(err, IsNil)
```

Comment:
We can use `defer` here, in case some error happens before `f.Close` and `os.Remove`.

```go
defer func() {
    f.Close()
    os.Remove(f.Name())
}()
```

Reply:
I suppose we should keep the file if some error happens so that we could look into the file to double check since this part of code is for test purpose. This piece of code is borrowed from previous code in `TestConfig`, which did not use `defer` neither.

Reply:
It'd be better use defer here to make the testing environment clean. TiDB has hundreds of contributors who may run the test but only few of them need to keep the file to do investigation. 
When you need double check you can delete the `os.Remove` code and rerun.

Reply:
The file is not deleted only if some test is broken, in this case manual inspection is needed (although it is not likely to be broken if the code is not changed), and you could/should manually delete that file, or rerun the test which will delete it if not thing went wrong. I agree it make sense to use defer here, I will change it if you insist.